soroban-env-common/src/symbol.rs

```diff
@@ -565,26 +567,23 @@ mod test_without_string {
         let input = "stellar";
         let sym = SymbolSmall::try_from_str(input).unwrap();
         let sym_str = SymbolStr::from(sym);
-        let s: &str = sym_str.as_ref();
-        assert_eq!(s, input);
+        assert_eq!(sym_str.to_string(), input);
```

Comment:
Why is this change necessary? Did we remove the as_ref() fn? I don't see it removed.

Reply:
I wanted to remove `as_ref`, but then we decided against that. I would probably still remove it though in order to reduce amount of unsafe code exported as it is not used in SDK and there is no way for the SDK user to get to `SymbolStr` (today it's probably fine, in 2 years someone introduces `SymbolStr` construction for whatever reason and suddenly `as_ref` becomes a footgun). In any case, even if we leave it, I would prefer to use the safer version in the env codebase.

Reply:
`SymbStr`'s `ToString` impl calls the `AsRef<str>` impl, so I don't think it's any safer as far as I can tell. For example:
https://github.com/stellar/rs-soroban-env/blob/75a838b4780890aa679bea3b6a47a0a029021d90/soroban-env-common/src/symbol.rs#L388-L391

Reply:
We could use the `&[u8]` version, but I've just reverted the changes for the sake of simplicity.